### ISSUE TITLE
Pattern match(es) are non-exhaustive when using Data.Text fields

### ISSUE THREAD

**Issue by pjones** [lykahb/groundhog#76]
I prefer that my code compile without any warnings.  If you point me in the right direction I'd be happy to submit a patch for this.  Thank you.

```
warning: [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a lambda abstraction:
        Patterns not matched:
            (Text.Read.Lex.Char _)
            (Text.Read.Lex.String _)
            (Text.Read.Lex.Punc _)
            (Text.Read.Lex.Ident [])
            ...
```

**Comment by lykahb** [comment]
Can you submit a small example that produces this error? I haven't seen it before.

**Comment by pjones** [comment]
Here you go:

https://gist.github.com/pjones/07a80e95d7d1cacdbdc07ff4d40587e5

**Comment by lykahb** [comment]
Fixed with b4ffe7b298a17446d577817bef5b44201a68ad59
